### PR TITLE
[16.0][ADD] project_tag_security: New module

### PR DESCRIPTION
**Pull request by victoralmau**
New module

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT51195

### CODE REVIEW

File: project_tag_security/views/project_project_views.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<odoo>
+    <record id="edit_project" model="ir.ui.view">
+        <field name="name">project.project.form</field>
+        <field name="model">project.project</field>
+        <field name="inherit_id" ref="project.edit_project" />
+        <field name="arch" type="xml">
+            <field name="tag_ids" position="attributes">
+                <attribute
+                    name="domain"
+                >['|', ('allowed_project_ids', 'in', [id]),('allowed_project_ids', '=', False)]</attribute>
```
Comment (pedrobaeza) [inline project_tag_security/views/project_project_views.xml:11]: Does this work?

```suggestion
                >[('allowed_project_ids', 'in', [False, id])]</attribute>
```

File: project_tag_security/views/project_tags_views.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            </field>
+        </field>
+    </record>
+    <record id="project_tags_form_view" model="ir.ui.view">
+        <field name="name">project.tags.form</field>
+        <field name="model">project.tags</field>
+        <field name="inherit_id" ref="project.project_tags_form_view" />
+        <field name="arch" type="xml">
+            <field name="color" position="after">
+                <field name="allowed_project_ids" widget="many2many_tags" />
```
Comment (pedrobaeza) [inline project_tag_security/views/project_tags_views.xml]: Put a placeholder saying "Empty for having all the projects allowed" (the same for the other if it's editable)

File: project_tag_security/views/project_project_views.xml (hunk 1 above)
Comment (victoralmau) [reply project_tag_security/views/project_project_views.xml:11]: I assumed you would suggest that but it doesn't work, with that domain you only get tags that have that project as allowed but not tags that have no project.

**Review by pedrobaeza — APPROVED** [review]

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Pull request merged**